src/deepsparse/loggers/metric_functions/cv/built_ins.py

```diff
@@ -0,0 +1,273 @@
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Dictionary that maps channel number to the std pixel value
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    stds = tuple(numpy.std(img_numpy, axis=tuple(dims)))
+    keys = ["std_channel_{}".format(i) for i in range(len(stds))]
+    return dict(zip(keys, stds))
+
+
+def fraction_zeros(img: Union[numpy.ndarray, "torch.tensor"]) -> float:  # noqa F821
+    """
+    Return the float the represents the fraction of zeros in the
+    image tensor/array
+
+    :param img: An image represented as a numpy array or a torch tensor.
+       Assumptions:
+           - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+             tensor/array
+           - the image has 3 or 1 channels
+    :return: A float in range from 0. to 1.
+    """
+    image_numpy = _assert_numpy_image(img)
+    _check_valid_image(image_numpy)
+    return (image_numpy.size - numpy.count_nonzero(image_numpy)) / image_numpy.size
+
+
+def count_classes_detected(
+    classes: List[List[Optional[Union[int, str]]]]
+) -> Dict[str, int]:
+    """
+    Count the number of unique classes detected in the image batch
+
+    :param classes: A nested list, where:
+        - first level is the batch information
+        - second level is the sample information. Can be
+            either `None` (no detection) or an integer/string
+            representation of the class label
+     :return: Dictionary, where the keys are class labels
+        and the values are their counts across the batch
+    """
+    _check_valid_classes(classes)
+    detected_classes = []
+    for sample_idx, sample in enumerate(classes):
+        if sample != [None]:
+            detected_classes += sample
+    counter = Counter(detected_classes)
+    # convert keys to strings if required
+    counter = {str(class_label): count for class_label, count in counter.items()}
+    return counter
+
+
+def count_number_objects_detected(
+    classes: List[List[Optional[Union[int, str]]]]
+) -> Dict[str, int]:
+    """
+    Count the number of successful detections per image
+
+     :param classes: A nested list, where:
+        - first level is the batch information
+        - second level is the sample information. Can be
+            either `None` (no detection) or an integer/string
+            representation of the class label
+    :return: Dictionary, where the keys are image indices within
+        a batch and the values are the number of detected objects per image
+    """
+    _check_valid_classes(classes)
+
+    number_objects_per_batch = {}
+    for sample_idx, sample in enumerate(classes):
```

Comment:
are `sample` and `batch` supposed to mean the same thing? if so, let's standardize to using only 1 for naming, otherwise, could use a comment to explain the difference